browser_patches/firefox/juggler/TargetRegistry.js

```diff
@@ -257,38 +257,55 @@ class TargetRegistry {
   }
 
   async newPage({browserContextId}) {
-    let window;
-    let created = false;
-    const windowsIt = Services.wm.getEnumerator('navigator:browser');
-    if (windowsIt.hasMoreElements()) {
-      window = windowsIt.getNext();
-    } else {
-      const features = "chrome,dialog=no,all";
-      const args = Cc["@mozilla.org/supports-string;1"].createInstance(Ci.nsISupportsString);
-      args.data = 'about:blank';
-      window = Services.ww.openWindow(null, AppConstants.BROWSER_CHROME_URL, '_blank', features, args);
-      created = true;
-    }
-    await waitForWindowReady(window);
     const browserContext = this.browserContextForId(browserContextId);
-    const tab = window.gBrowser.addTab('about:blank', {
-      userContextId: browserContext.userContextId,
-      triggeringPrincipal: Services.scriptSecurityManager.getSystemPrincipal(),
-    });
-    const target = await new Promise(fulfill => {
+    const features = "chrome,dialog=no,all";
+    // See _callWithURIToLoad in browser.js for the structure of window.arguments
+    // window.arguments[1]: unused (bug 871161)
+    //                 [2]: referrerInfo (nsIReferrerInfo)
+    //                 [3]: postData (nsIInputStream)
+    //                 [4]: allowThirdPartyFixup (bool)
+    //                 [5]: userContextId (int)
+    //                 [6]: originPrincipal (nsIPrincipal)
+    //                 [7]: originStoragePrincipal (nsIPrincipal)
+    //                 [8]: triggeringPrincipal (nsIPrincipal)
+    //                 [9]: allowInheritPrincipal (bool)
+    //                 [10]: csp (nsIContentSecurityPolicy)
+    //                 [11]: nsOpenWindowInfo
+    const args = Cc["@mozilla.org/array;1"].createInstance(Ci.nsIMutableArray);
```

Comment:
simple array wouldn't work?

Reply:
No :smile: Apparently JS array breaks when this is passed to native code. This is what people do in other places in Firefox codebase.